### PR TITLE
Replace device-info permission by a browsing context boolean flag

### PR DESCRIPTION
**Pull request by youennf**

### CODE REVIEW

**Comment by alvestrand** [comment]
I disagree with this approach.
I think it has no advantages over the permission-based approach, and requires special-casing situations where we want to have device-enumeration permission separate from device-access permission, rather than just saying "and you can grant this permission in that case too".

Please file a bug describing the problem you want to solve.

**Comment by youennf** [comment]
The bug is https://github.com/w3c/mediacapture-main/issues/612.
It is not great that because I give access to a camera once to take a picture of me, the website will be able to get all my device information for my future visits.

Also, we agreed that the usual good approach is to do getUserMedia-then-enumerateDevices.
This PR is consistent with that approach.

Additionally, I am not sure the device info permission concept is good.
Other permissions are things like "can I use a camera", which is easy to describe to a user.
This specific permission is hard to explain to a user and AFAIK, is not exposed by any UI, contrary to camera/mic access.

**Comment by jan-ivar** [comment]
This PR removes an arguably redundant permission, which I approve of.

But it still says user agents may revoke this ability (permission) _whenever they want_. This could be:
1. Never
2. During actual camera or mic capture, sabotaging post-gum in-content selection 

I think we need to define this in more detail to protect web interop. 

Original intent: when users revoke both camera and microphone permission, they revoke device-info as well (not before). I.e.
```
device_info = camera_permission || microphone_permission
```
This is true in all browsers I tested, so it must have been the original intent. Other parts like `groupId` support this notion. I vote we tighten the spec to conform with implementations here.

This was also meant to be extensible, which is where I share @alvestrand's concern. E.g.
```
device_info = camera_permission || microphone_permission || speakers
```
If we end up not needing this because of https://github.com/w3c/mediacapture-output/pull/86, great! But it would still be nice if this spec didn't forbid extension, I think—Forbidding user agents from doing whatever they want is not the same as forbidding other specs from extending things.

Is there a way to write it like this?

**Comment by jan-ivar** [comment]
I understand permissions may be per-device, so if this needs to be per-device that would WFM.

File: getusermedia.html [hunk 1]
```diff
@@ -3107,18 +3109,18 @@ <h2>Device information can be exposed check</h2>
...
-          <li>If any of the input devices are attached to an
-            <a href="#stream-active">active</a> <code><a>MediaStream</a></code>
-            in the browsing context, return <code>true</code>.</li>
           <li>If the <a>current settings object</a>'s <a>responsible
           document</a> is <a data-cite="!HTML/#fully-active">fully
           active</a> and <a data-cite="!HTML/#gains-focus">has
           focus</a>, return <code>true</code>.
           </li>
           <li>Return <code>false</code>.</li>
+          <div class="note">
```
Comment (dontcallmedom) [inline getusermedia.html]: this would need to be moved in the `<li>`

**Comment by alvestrand** [comment]
Needs a PR to remove device-info from feature-policy. @youennf has promised to do this.

**Pull request merged**